glusterd2/commands/volumes/volume-expand-txn.go

```diff
@@ -40,12 +40,14 @@ func expandValidatePrepare(c transaction.TxnCtx) error {
 		return errors.New("invalid number of bricks")
 	}
 
-	if volinfo.Type == volume.Replicate && req.ReplicaCount != 0 {
-		// TODO: Only considered first sub volume's ReplicaCount
-		if req.ReplicaCount < volinfo.Subvols[0].ReplicaCount {
-			return errors.New("invalid number of bricks")
-		} else if req.ReplicaCount == volinfo.Subvols[0].ReplicaCount {
-			return errors.New("replica count is same")
+	if volinfo.Type == volume.Replicate || volinfo.Type == volume.DistReplicate {
+		if req.ReplicaCount != 0 {
+			// TODO: Only considered first sub volume's ReplicaCount
+			if req.ReplicaCount < volinfo.Subvols[0].ReplicaCount {
+				return errors.New("invalid number of bricks")
+			} else if req.ReplicaCount == volinfo.Subvols[0].ReplicaCount {
```

Comment:
if the replica count is the same as it was, then we are supposed to add a new subvol aren't we?

Reply:
usability issue if we add new sub volume. If user specified replica count with some expectation, if we add sub volume instead of notifying user then it will be wrong.